modules/gcloud/src/test/scala/com/dimafeng/testcontainers/PubSubEmulatorContainerSpec.scala

```diff
@@ -0,0 +1,105 @@
+package com.dimafeng.testcontainers
+
+import com.google.cloud.pubsub.v1.{AckReplyConsumer, MessageReceiver}
+import com.google.protobuf.ByteString
+import com.google.pubsub.v1._
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import java.util.concurrent.TimeUnit
+import scala.collection.JavaConverters._
+import scala.concurrent.TimeoutException
+
+class PubSubEmulatorContainerSpec extends AnyWordSpecLike with Matchers with ForAllTestContainer {
+
+  override val container: PubSubEmulatorContainer = PubSubEmulatorContainer()
+
+  override def beforeStop(): Unit = {
+    container.topicAdminClient.close()
```

Comment:
I'm curious if `container.topicAdminClient.close()` and `container.subscriptionAdminClient.close()` can be done in the `container.close()`

Reply:
Good point. I changed the logic so it's done in `PubSubEmulatorContainer` before closing the container.